app/controllers/api/v0/search_controller.rb

```diff
@@ -1,47 +1,64 @@
-class Api::V0::SearchController < ApplicationController
+class Api::V0::SearchController < Api::V0::BaseController
 
-  def search
-    # TODO generalize
-    query = {
-      "size": 25,
-      "query": {
-        "multi_match": {
-          "query": params['q']
-          # "fields": ["fields.title^4", "fields.plot^2", "fields.actors", "fields.directors"]
-        }
-      },
-      "_source": ["id"],
-       "highlight": {
-          "fields": {
-            "content": {}
-          }
-        }
-    }
-
-    response = OpenSearch::ElasticsearchClient.instance.search body: query.to_json
-    render json: response
+  swagger_path '/search' do
+    operation :get do
+      key :summary, 'Run a search query'
+      key :description, 'Run a search query'
+      key :operationId, 'search'
+      parameter do
+        key :name, :q
+        key :in, :query
+        key :description, 'text of the search query'
+        key :required, true
+        key :type, :string
+      end
+      parameter do
+        key :name, :book
+        key :in, :query
+        key :description, 'UUID@version of the book to search'
+        key :required, true
+        key :type, :string
+      end
+      parameter do
+        key :name, :index_strategy
```

Comment:
didn't we discuss this being fixed per api version and the client not needing to care about it?

Reply:
Let's maybe park this.  I do recall some conversation related to this, but there are reasons to not have a different API version for every index strategy / search strategy combination.  As it is currently...

* The API version controls the format of the requests and the top-level format of the response
* The index strategy controls what is indexed and how it is indexed (e.g. page elements, indexing options, etc)
* The search strategy controls how we search the index (fuzzy vs not fuzzy, boosting fields, full matches vs OR matches, etc)

The API is returning raw ES results plus some top-level metadata.  The top-level metadata is fixed for each API version.  The raw ES results depend on the combination of strategies.

One API version, defining the request format and top-level response format, could work with several combinations of strategies.  

There likely won't be a zillion strategies, and the Rex FE would change slowly over time to use different ones (e.g. the `i1` indexing strategy could be used for many months).  

The `search_strategy` parameter would give us a way to A/B test different search queries (with the A/B machinery being external to search itself).

Reply:
i was thinking that indexing/search strategy changes that did not influence the structure of the responses could be done within one api version, and any other change would create an api version. exposing 3 versions of things for the client to manage seems like a lot, and i was actually thinking it would be nicer to handle a/b testing of search stuff within search, rather than forcing the client to deal with all that

Reply:
After a discussion with Tom, we decided it is ok for the time being to leave the query parameters that choose index strategy and search strategy.